src/charm_state.py

```diff
@@ -618,40 +617,20 @@ class ProxyConfig(BaseModel):
     no_proxy: Optional[str]
     use_aproxy: bool = False
 
-    @classmethod
-    def from_charm(cls, charm: CharmBase) -> "ProxyConfig":
-        """Initialize the proxy config from charm.
-
-        Args:
-            charm: The charm instance.
-
-        Returns:
-            Current proxy config of the charm.
-        """
-        use_aproxy = bool(charm.config.get(USE_APROXY_CONFIG_NAME))
-        http_proxy = get_env_var("JUJU_CHARM_HTTP_PROXY") or None
-        https_proxy = get_env_var("JUJU_CHARM_HTTPS_PROXY") or None
-        no_proxy = get_env_var("JUJU_CHARM_NO_PROXY") or None
-
-        # there's no need for no_proxy if there's no http_proxy or https_proxy
-        if not (https_proxy or http_proxy) and no_proxy:
-            no_proxy = None
-
-        return cls(
-            http=http_proxy,
-            https=https_proxy,
-            no_proxy=no_proxy,
-            use_aproxy=use_aproxy,
-        )
-
     @property
     def aproxy_address(self) -> Optional[str]:
         """Return the aproxy address."""
         if self.use_aproxy:
             proxy_address = self.http or self.https
             # assert is only used to make mypy happy
-            assert proxy_address is not None  # nosec for [B101:assert_used]
-            aproxy_address = f"{proxy_address.host}:{proxy_address.port}"
+            assert (
```

Comment:
Fixed bug where if port is not defined, address would come out as <Address>:None